### PR TITLE
Only generate keystore.xml once

### PR DESCRIPTION
**Pull request by davidcurrie**
Fix for https://github.com/WASdev/ci.docker/issues/68 including new test case (and refactoring of existing testcases)

### CODE REVIEW

File: websphere-liberty/test/build.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+test=test
+cname=$tag$test
+arch=`uname -p`
+
+if [ $# != 2 ]
+then
+   if [ $# != 1 ]
+   then
+      echo "Usage : buildAndVerify.sh <Image name> <Dockerfile location>"
```
Comment (liamawhite) [inline websphere-liberty/test/build.sh]: this should be build.sh? <Image name> <Dockerfile location>

File: websphere-liberty/test/build.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
+#! /bin/bash
+#####################################################################################
+#                                                                                   #
+#  Script to build a docker image                                                   #
+#                                                                                   #
+#                                                                                   #
+#  Usage : build.sh <Image name> <Dockerfile location         >                     #
```
Comment (liamawhite) [inline websphere-liberty/test/build.sh]: lots of space between location and >

**Comment by liamawhite** [comment]
@davidcurrie looks good, just a few cosmetic things.

**Comment by davidcurrie** [comment]
Thanks @liamawhite. I've updated those and managed another green build so please merge if you're happy...

**Pull request merged**